language-design.md

```diff
@@ -16,19 +16,78 @@ are:
+[Steele, 1998].
+
+This quote contrasts with the ideas that we discussed in class, specifically that languages cycle
+through developing into a DSL, then into a general purpose language, and then back and through
+the cycle. PL/I was designed to not grow, which is a different philosophy from the other
+languages we have encountered. It would be interesting to discover what drove the design 
+philosophy for PL/I because it is such a large, immutable language.
+
+> "A language design can no longer be a thing. It must be a pattern—a pattern for growth—a 
+> pattern for growing the pattern for defining the patterns that programmers can use for 
+> their real work and their main goal"
+[Steele, 1998].
+
+By thinking of language as a pattern, Steele turns the focus from the programmer to the user.
+By understanding the pattern, users can expand the language, which Steele calls "a pattern for
+growth." At the very least, users feel like they are contributing to an ongoing project as
+opposed to using the results of a finished project. As a bonus, user feedback and suggestions 
+can facilitate a language's growth. The difficulty for programmers is not being able to consider
+what they would like to see, but considering what a user of their language would like to see.
 
 ---
-
 **Question**
 
 How would you know a well-designed language? What are the symptoms? How would
 you know a poorly designed language? What are the symptoms?
 
 **Response**
-
-
+A well-designed language can grow into your project. If you have an idea, a well-designed 
+language will have a way to help you achieve that vision, while a 
+poorly-designed language forces you to significantly change your idea to adapt to what the
+language can accomplish. One thing we disagree on is whether a well-designed language must 
+always have a large, helpful community. Sam thinks that a community focused on solving user problems
+can help languages grow since they should please the users. But Tiffany thinks that a large community only 
+reflects how widespread a language is. For example, JavaScript has a large community, but 
+many people claim it is poorly designed. 
+
+On the other hand, the community can be an indicator of how well or poorly designed a language is. 
+Tiffany found an article named ["The Python documentation is bad, and you should feel bad"](http://cryto.net/~joepie91/blog/2013/02/19/the-python-documentation-is-bad-and-you-should-feel-bad/). 
+According to the article, while the Python community is rather large, it does encourage a 
+"Just read the source" mentality over meaningful documentation (and the author continues on  
+saying how bad Python's documentation is). Although Python can grow, its user base encourages 
+poor practices, especially telling people to look at poor documentation instead of helping them 
+with their specific problems. With these users controlling the online feedback to Python, 
```

Comment:
I think part of the difference (and you'll see this a lot on Stack Overflow) is that some people will just dump a bunch of code (or just say, "I want to do X. How?") and ask for help debugging, which doesn't really incline people to help (especially considering that many of these questions are best answered by "Learn to program, and then read the documentation").